apps/dashboard/src/components/workflow-editor/steps/common-fields.tsx

```diff
@@ -1,12 +1,33 @@
+import { useLayoutEffect, useState } from 'react';
+import { useNavigate, useParams } from 'react-router-dom';
 import { useWorkflowEditorContext } from '@/components/workflow-editor/hooks';
 import { CopyButton } from '../../primitives/copy-button';
 import { FormControl, FormField, FormItem, FormLabel, FormMessage } from '../../primitives/form/form';
 import { Input, InputField } from '../../primitives/input';
 import { useStep } from './use-step';
+import { buildRoute, ROUTES } from '@/utils/routes';
 
 export function CommonFields() {
-  const { stepIndex, control } = useStep();
+  const { stepIndex, control, step } = useStep();
+  const navigate = useNavigate();
+  const { stepSlug } = useParams<{ stepSlug: string }>();
   const { isReadOnly } = useWorkflowEditorContext();
+  const [isBlurred, setIsBlurred] = useState(false);
+
+  const isStepSlugChanged = step && step?.slug && stepSlug !== step.slug;
+  const shouldUpdateStepSlug = isBlurred && isStepSlugChanged;
+
+  useLayoutEffect(() => {
+    if (shouldUpdateStepSlug) {
+      setTimeout(() => {
+        navigate(buildRoute(`../${ROUTES.CONFIGURE_STEP}`, { stepSlug: step?.slug ?? '' }), {
+          replace: true,
+          state: { skipAnimation: true },
+        });
+      }, 0);
+      setIsBlurred(false);
+    }
+  }, [shouldUpdateStepSlug, step, navigate]);
```

Comment:
When the name of the step is updated, the `stepSlug` in the URL becomes outdated. It is updated to the latest value when the `name` field loses focus (blurred).